src/MultiRegion/cubed_sphere_boundary_conditions.jl

```diff
@@ -32,39 +33,81 @@ function fill_halo_regions!(field::CubedSphereField{<:Center, <:Center})
     Hx == Hy || error("horizontal halo size Hx and Hy must be the same")
     Hc = Hx
 
-    #-- one pass: only use interior-point values:
-    for region in 1:6
+    multiregion_field = Reference(field.data.regional_objects)
+    region = Iterate(1:6)
 
-        region_E, region_N, region_W, region_S = find_neighboring_panels(grid, region)
+    kernel_parameters = KernelParameters((Hc, Nc, Nz), (0, 0, 0))
+    @apply_regionally begin
+        launch!(grid.architecture, grid, kernel_parameters, _fill_cubed_sphere_center_center_field_east_west_halo_regions!,
+                field, multiregion_field, n_regions, region, Hc, Nc)
+    end
```

Comment:
```suggestion
    @apply_regionally begin
        launch!(grid.architecture, grid, (Hc, Nc, Nz), _fill_cubed_sphere_center_center_field_east_west_halo_regions!,
                field, multiregion_field, n_regions, region, Hc, Nc)
    end
```

Reply:
It turns out I actually need the offset in the z direction for multi-layer halo fillings.